lib/spack/spack/database.py

```diff
@@ -698,7 +706,7 @@ def query_one(self, query_spec, known=any, installed=True):
         query. Returns None if no installed package matches.
 
         """
-        concrete_specs = self.query(query_spec, known, installed)
+        concrete_specs = self.query(query_spec, known, installed, unique=True)
         assert len(concrete_specs) <= 1
```

Comment:
I put @tgamblin as a reviewer because skimming the code `query_one` seems to be used only in tests. In that case it may be fine to erase it completely, and have a `unique` argument to `query`. 

The only comment that would remain from my side is that I wouldn't use `tty` directly in the `database` module, but raise an appropriate error. The command should then catch it and react with `tty.die`.